### PR TITLE
Add DeprecationHelper DeprecatedScopeResolver

### PR DESCRIPTION
**Pull request by mglaman**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,9 +11,9 @@
     ],
     "require": {
         "php": "^7.4 || ^8.0",
-        "symfony/finder": "^4.2 || ^5.0 || ^6.0 || ^7.0",
-        "phpstan/phpstan": "^1.10.1",
+        "phpstan/phpstan": "1.10.x-dev",
```
Comment (mglaman) [inline composer.json:14]: Will mark ready once https://github.com/phpstan/phpstan-src/commit/b87e5c4c7e33a91c61341f0335a221c32df603b2 is released

**Comment by bbrala** [comment]
This is great, guess its time to start spinning up the d11 readiness bot :)

**Comment by bbrala** [comment]
New release was published: https://github.com/phpstan/phpstan/releases/tag/1.10.56

**Comment by mglaman** [comment]
Updated to require PHPStan ^1.10.56

**Pull request merged**